### PR TITLE
Add string check for enums

### PR DESCRIPTION
**Pull request by Naragato**
#459

### CODE REVIEW

**Comment by Naragato** [comment]
Actually slight error in my PR will correct now :)

**Comment by bgrainger** [comment]
Please add a test that verifies the new behaviour.

**Comment by Naragato** [comment]
Done. : >

**Comment by bgrainger** [comment]
Sorry, I wasn't clear: the test needs to be in the SideBySide project to verify that the behaviour is the same in MySqlConnector and Connector/NET (since this is trying to clone existing Connector behaviour).

Please also revert the changes to the key files.

**Comment by Naragato** [comment]
Reverted the certs (not sure what actually changed them :/ I left the original test in place as it makes sense to be there. However I added an additional test into the sidebyside project.

File: src/MySqlConnector/MySql.Data.MySqlClient/MySqlParameter.cs [hunk 1]
```diff
@@ -303,6 +303,10 @@ internal void AppendSqlString(BinaryWriter writer, StatementPreparerOptions opti
 			{
 				writer.WriteUtf8("{0}".FormatInvariant((ulong) Value));
 			}
+			else if (MySqlDbType == MySqlDbType.String || MySqlDbType == MySqlDbType.VarChar && HasSetDbType && Value is Enum)
```
Comment (bgrainger) [inline src/MySqlConnector/MySql.Data.MySqlClient/MySqlParameter.cs]: Precedence requires parentheses around `(MySqlDbType == MySqlDbType.String || MySqlDbType == MySqlDbType.VarChar)`.

**Comment by bgrainger** [comment]
Thanks for the bug report and PR!

**Pull request merged**